ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -166,7 +172,19 @@ private void initSpatials() {
         levelGenerator = new LevelGenerator(WORLD_LOCATION);
         platform = new Platform(PLATFORM_LOCATION);
         commander = new Commander(COMMANDER_LOCATION);
-
+        carriers = new Carrier[4];
+        for (int i = 0; i < carriers.length; i++) {
+            float x = RELATIVE_CARRIER_LOCATION.x;
+            float z = RELATIVE_CARRIER_LOCATION.z;
+            float y = RELATIVE_CARRIER_LOCATION.y;
+            if ((i + 1) % 2 == 0) {
+                z =  -z;
+            }
+            if (i > 1) {
+                x = -x;
+            }
+            carriers[i] = new Carrier(COMMANDER_LOCATION.add(new Vector3f(x, y, z)), i);
+        }
```

Comment:
Maybe put this in a seperate method createCarriers that returns a Carrier array. Then call method in initSpatials by using carriers = createCarriers();

Also, initialize your float x,z,y outside of the for loop .

It also would be nice for there to be comment in the `if ((i + 1) % 2 == 0) ` and `if (i > 1)`, because I don't quite understand what these do.